### DISCUSSION TITLE
Use python code as plugin

### DISCUSSION THREAD

**Discussion by GuillaumeTh** [apache/superset#19205] [Q&A / Help]
I would like to know if it is possible to use python code as plugin instead of TypeScript code.

As an example, I would like to perform a pearson correlation. I already have this code on my hand but I would like to use it with superset interface. Is it possible to do that ? If yes do you have a documentation about that ?

Thanks

Guillaume

**Comment by ktmud** [comment]
Currently you can add [Pandas post processing](https://github.com/apache/superset/blob/4d329071a1651994574c3ad1f26e32b56d9ff810/superset/utils/pandas_postprocessing.py#L0-L1) for certain charts (those using the new API endpoint), but not all of them.

Although for your specific case, Pearson correlation is also very easy to calculate [in JavaScript](https://gist.github.com/matt-west/6500993). For reference, we do have a [Paired t-test chart](https://superset-ui.superset.vercel.app/?path=/story/legacy-chart-plugins-legacy-plugin-chart-paired-t-test--basic) maybe you can extend from.

**Comment by villebro** [comment]
@GuillaumeTh I have actually created a viz that's able to run Python code and render just like regular charts, but it's not quite ready for public consumption yet. In the meantime, would you be able to shar what your preferred workflow would look like? I assume you would like to be able to write regular python code, much like in a notebook, and then render a matplotlib or similar chart in the chart window?

**Reply by micaelfilipee** [reply]
@villebro I'm interested in the plugin you're developing. Would it be possible for you to share the code so I can try it out? Thank you!

**Comment by GuillaumeTh** [comment]
@ktmud Thanks I will check that. In our case Pearson is a small example. We have more complex stats already in python. @villebro currently I have my own Flask server and I get infos from html dropdown. After that a python code run pearson correlation on selected data and display table and graphs in the HTML Flask page. Is it fit in your current workflow ?

**Comment by aliJabra** [comment]
Going down the same route, I found these [SQL based functions](https://www.red-gate.com/simple-talk/blogs/statistics-sql-pearsons-correlation/) useful for 2d series, but I'm looking forward to @villebro's python-bridge because nothing beats pandas.corr() s simplicity in this case.

**Comment by GuillaumeTh** [comment]
Hi @villebro I am interesting to test your python support and give you feed back if you want. This python support is important for us and could be a game changer.

**Comment by fazilhero** [comment]
Any updates on this discussion?
